core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
+  async deleteExploration(): Promise<void> {
+    await this.clickOn(deleteExplorationButton);
+    await this.clickOn(deleteConfirmButton);
   }
 
   /**
-   * Function to publish exploration
+   * This function save the changes of the exploration.
    */
-  async publishExplorationWithTitle(
-    explorationTitle: string
-  ): Promise<string | null> {
+  async saveDraftExploration(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileSaveDraftButton);
+    } else {
+      await this.clickOn(desktopSaveDraftButton);
+    }
+
+    await this.clickOn(commitMessage);
+    await this.type(commitMessage, 'Testing Testing');
+    await this.clickOn(saveDraftConfirmButton);
+    await this.page.waitForFunction('document.readyState === "complete"');
+    showMessage('Exploration is saved successfully.');
+  }
+
+  async publishExploration(): Promise<string | null> {
+    await this.saveDraftExploration();
     if (this.isViewportAtMobileWidth()) {
       await this.page.waitForSelector('.e2e-test-toast-message', {
         visible: true,
       });
       await this.page.waitForSelector('.e2e-test-toast-message', {
         hidden: true,
       });
-      await this.clickOn(mobileChangesDropdown);
+      await this.clickOn(publishButtonDropDown);
       await this.clickOn(mobilePublishButton);
     } else {
-      await this.page.waitForSelector(
-        `${publishExplorationButton}:not([disabled])`
-      );
-      await this.clickOn(publishExplorationButton);
-    }
-    await this.clickOn(explorationTitleInput);
-    await this.type(explorationTitleInput, `${explorationTitle}`);
-    await this.clickOn(explorationGoalInput);
-    await this.type(explorationGoalInput, `${explorationTitle}`);
-    await this.clickOn(explorationCategoryDropdown);
-    await this.clickOn('Algebra');
-    await this.clickOn(saveExplorationChangesButton);
-    await this.clickOn(explorationConfirmPublishButton);
-    await this.page.waitForSelector(explorationIdElement);
-    const explorationIdUrl = await this.page.$eval(
-      explorationIdElement,
-      element => (element as HTMLElement).innerText
-    );
-    const explorationId = explorationIdUrl.replace(/^.*\/explore\//, '');
+      await this.clickOn(desktopPublishButton);
+    }
+    await this.clickOn(confirmPublishButton);
+    await this.page.waitForSelector(closePublishedPopUpButton, {visible: true});
+
+    explorationUrlAfterPublished = await this.page.url();
+    const explorationId = explorationUrlAfterPublished
+      .replace(/^.*\/create\//, '')
+      .replace(/#\/.*/, '');
+
+    await this.clickOn(closePublishedPopUpButton);
+    await this.expectExplorationToBePublish();
 
-    await this.clickOn(closeShareModalButton);
     return explorationId;
   }
 
+  async expectExplorationToBeAccessibleWithTheUrl(
+    expression: string
+  ): Promise<void> {
+    if (expression === 'Yes') {
+      try {
+        await this.page.goto(explorationUrlAfterPublished);
```

Comment:
Don't you need to pass the explorationId into the function so that you can construct the exploration URL? Things specific to the top-level test should not be hardcoded into the utility functions.